.github/workflows/promptflow-core-test.yml

```diff
@@ -71,7 +71,7 @@ jobs:
         run: |
           cp ${{ github.workspace }}/src/promptflow/dev-connections.json.example ${{ github.workspace }}/src/promptflow/connections.json
       - name: run core tests
-        run: poetry run pytest ./tests/core --cov=promptflow --cov-config=pyproject.toml --cov-report=term --cov-report=html --cov-report=xml
+        run: poetry run pytest ./tests/core --cov=promptflow --cov-config=pyproject.toml --cov-report=term --cov-report=html --cov-report=xml --tb=short
```

Comment:
may add how `--tb=short` behaves in the PR description

Reply:
also, will `short` resolve the leak issue? shall we use `no`?

Reply:
Let's try short first https://docs.pytest.org/en/latest/how-to/output.html#modifying-python-traceback-printing

Reply:
verified, `short` works well

![image](https://github.com/microsoft/promptflow/assets/38847871/224dc08d-a819-4cd2-a99c-cb072f7c53ac)